sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -91,13 +91,25 @@ case class CreateDataSourceTableCommand(table: CatalogTable, ignoreIfExists: Boo
       table.storage.properties
     }
 
+    val newProps = if (partitionColumnNames.nonEmpty &&
+        sparkSession.sqlContext.conf.manageFilesourcePartitions) {
+      // Start off with partition provider hive, but no partitions in the metastore. The user
+      // has to call `msck repair table` to populate the table partitions.
+      table.properties ++
+        Map(CatalogTable.PARTITION_PROVIDER_KEY -> CatalogTable.PARTITION_PROVIDER_HIVE)
```

Comment:
Can users set/unset this special property? I guess no? and `DESC TABLE`, `SHOW CREATE TABLE` should also exclude it right?